File: Python/1502.CanMakeArithmeticProgressionFromSequence.py
#Have to figue out if the given list is in arithmetic progression or not.
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr.sort()
        dif=arr[1]-arr[0]
        for i in range (2,len(arr)):
            d=arr[i]-arr[i-1]
            if d!=dif:
                return False
        return True    

File: C++/1502.CanMakeArithmeticProgressionFromSequence.cpp
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        int n = arr.size();

        //since, there is no 3rd element to compare
        if(n==2)
            return true;
        sort(begin(arr), end(arr));
        for(int i=0; i<n-2; i++)
        {
            //checking the difference between any 2 consecutive elements
            if(arr[i] - arr[i+1] != arr[i+1] - arr[i+2])
                return false;
        }
        return true;
    }
};
